libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -773,7 +771,13 @@ public void onResponse(ImageLoader.ImageContainer container, boolean isImmediate
 
             addDefaultSizeClassIfMissing(attrs);
 
-            Bitmap bitmapToShow = ImageUtils.getWPImageSpanThumbnailFromFilePath(getActivity(), safeMediaPreviewUrl, maxWidth);
+            int[] bitmapDimensions = ImageUtils.getImageSize(Uri.parse(safeMediaPreviewUrl), getActivity());
+            int realBitmapWidth = bitmapDimensions[0];
+            Bitmap bitmapToShow = ImageUtils.getWPImageSpanThumbnailFromFilePath(
+                    getActivity(),
+                    safeMediaPreviewUrl,
+                    maxWidth > realBitmapWidth && realBitmapWidth > 0 ? realBitmapWidth : maxWidth);
```

Comment:
Do you think we should enforce a _minimum_ width, to avoid users adding a tiny image that they then have trouble removing?

Reply:
As a follow-up on the above, it looks like we define a minimum size [here](https://github.com/wordpress-mobile/WordPress-Android/blob/e64e9f5f1ce79cb3764813d1e8150d75b508e3a2/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java#L108) for use with media library images. Perhaps we should do the same for device images?

Reply:
This is an interesting question. I thought about it from this angle - we want to show content in the editor as close as possible to what it will look like on the post/page. It works the same way in calypso. When someone uploads a tiny image, and we scale it up, it gives the impression that "Oh. Doesn't look so bad", when in reality the image will be tiny.

But the inability to easily tap on a small image is not cool (it's easier to do with a mouse in Calypso), so I'll look into what minimum size will work the best for us. Maybe setting it to a line height will do the trick.

Reply:
I tried to set the min size to a line height, and it works beautifully, but to get remote images to load with min size, I need to update some code on the Aztec side, which will take a bit of time.

Reply:
We can tackle the min size a separate PR and go ahead and merge this one, if you're okay with that.

Reply:
Yes, this would be great. This way the image will at least look same be it right after attachment or later when loading it from URL.